pkg/virtual/framework/forwardingregistry/rest.go

```diff
@@ -85,3 +89,64 @@ func NewStorage(ctx context.Context, resource schema.GroupVersionResource, apiEx
 	statusStore := wrapper(resource.GroupResource(), statusDelegate)
 	return store, statusStore
 }
+
+// ProvideReadOnlyRestStorage returns a commonly used REST storage that forwards calls to a dynamic client,
+// but only for read-only requests.
+func ProvideReadOnlyRestStorage(ctx context.Context, clusterClient dynamic.ClusterInterface, wrapper StorageWrapper) (apiserver.RestProviderFunc, error) {
+	return func(resource schema.GroupVersionResource, kind schema.GroupVersionKind, listKind schema.GroupVersionKind, typer runtime.ObjectTyper, tableConvertor rest.TableConvertor, namespaceScoped bool, schemaValidator *validate.SchemaValidator, subresourcesSchemaValidator map[string]*validate.SchemaValidator, structuralSchema *structuralschema.Structural) (mainStorage rest.Storage, subresourceStorages map[string]rest.Storage) {
+		statusSchemaValidate := subresourcesSchemaValidator["status"]
+
+		strategy := customresource.NewStrategy(
+			typer,
+			namespaceScoped,
+			kind,
+			schemaValidator,
+			statusSchemaValidate,
+			map[string]*structuralschema.Structural{resource.Version: structuralSchema},
+			nil, // no status here
+			nil, // no scale here
+		)
+
+		storage, _ := NewStorage(
+			ctx,
+			resource,
+			"", // ClusterWorkspaces have no identity
```

Comment:
So far this is used for APIExports, not ClusterWorkspaces?

Reply:
oh i see this got moved from the initializingworkspaces package. can you update this comment?

Reply:
to be honest, I don't understand even the old comment. But yes, just moved.

Done (removed).